### PR TITLE
added preprocessing code

### PR DESCRIPTION
**Pull request by HrishikeshGeedMS**
# Description
Objective was to add the preprocessing node/component in automl-pipeline examples, which user can use for preprocessing the data set before training.
We have added the frame work for sdk-cli pipelines code, user can change it as per the need.

Preprocessing components are added here in following pipeline examples

- sdk/python/jobs/pipeline/1h_automl_in_pipeline
- cli/jobs/pipelines/automl

Created new branch and copied the code from hrishikesh/preprocessing_node
# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by novaturient95 — APPROVED** [review]

File: cli/jobs/pipelines/automl/forecasting-energy-demand-pipeline/training-mltable-folder/MLTable [hunk 1]
```diff
@@ -4,10 +4,10 @@ transformations:
...
         encoding: 'ascii'
-  - convert_column_types:
-      - columns: demand
-        column_type: float
-      - columns: precip
-        column_type: float
-      - columns: temp
-        column_type: float
\ No newline at end of file
+  # - convert_column_types:
```
Comment (novaturient95) [inline cli/jobs/pipelines/automl/forecasting-energy-demand-pipeline/training-mltable-folder/MLTable:7]: revert?

**Comment by skasturi** [comment]
Could you please check the build failures?

**Comment by skasturi** [comment]
Discussed with @santiagxf , @SagarikaKengunte offline and the build failures are being fixed by @santiagxf . 

image

**Review by skasturi — APPROVED** [review]

**Comment by skasturi** [comment]
Please add more details to the description.

**Pull request merged**